### DISCUSSION TITLE
Scaling jobs horizontally on ECS

### DISCUSSION THREAD

**Discussion by axelwas** [dagster-io/dagster#18246] [Q&A] [answered]
Hi all,

I am quite new to Dagster and wonder as to how to best implement one of our use cases with the right Dagster abstractions.

We need a workflow that would allow us to refresh a Tableau dashboard, using Tableau datasources, which point to Snowflake tables tables downstream of a dbt pipeline and a bunch of ingestion pipelines.
As I understand, with Dagster native integrations and some custom Python code, I can generate  for each Tableau dashboard the following lineage :
`source assets -> dbt assets -> Tableau datasource assets -> Tableau dashboard asset`
I would then create a job with define_asset_job() , and select the Tableau asset key and its upstream assets :
```python
my_tableau_dashboard_job = define_asset_job(
    "my_tableau_dashboard_job",
    AssetSelection.keys(AssetKey(["dashboards", "my_tableau_dashboard"]))
    .upstream()
    .required_multi_asset_neighbors(),
)
```

I am currently running Dagster OSS on ECS. I wonder about the limitations of the deployment to scale horizontally within this kind of job. Indeed, I would ideally need to run each step in a dedicated ECS task. The reason for that is some custom Python source assets are very memory/compute intensive, while the dbt and Tableau assets aren’t at all.

Obviously I cannot use any of the k8s executors to run this job. Would it be possible to use the [docker_executor](https://docs.dagster.io/_apidocs/libraries/dagster-docker#dagster_docker.docker_executor) and have it spin up an ECS task for each asset within the job ?
If it is not possible, is there any other way to split the big job into smaller ones and somehow create a dependency between them so that, for example, a job that materializes the tableau dashboard asset needs to trigger other upstream jobs (in dedicated ECS tasks) and wait for completion before running itself ?

Thanks for your help !

**Comment by gibsondan** [comment, marked as answer]
There's a feature request issue for creating an ECS task executor, which would allow you to run each asset within a job in its own ECS task: https://github.com/dagster-io/dagster/issues/9671 - that's not currently available though.

If using kubernetes instead of ECS is an option, the k8s_job_executor in k8s allows you to run each op in its own kubernetes job, with different resource limits.

Finally, you could also have an op that runs an ECS taks using the boto API, but that wouldn't give you any of Dagster's built-in observability benefits or programming model.

 Dagster Pipes (https://docs.dagster.io/guides/dagster-pipes) can also give you some more options for running compute in external environments, even if your op is still running within a shared ECS task for the run.

**Reply by axelwas** [reply]
Hi @gibsondan,

Thanks for the thorough answer. That confirms my concerns about execution on ECS being limited when it comes to job scaling.

Regarding run status sensors, I was thinking about a sensor that would apply the rule "whenever a new execution of jobs A, B and C have succeeded, run job D". However, looking at the codebase my understanding is that by specifying jobs A, B, C in the `monitored_jobs` parameter of `@run_status_sensor`, the sensor will trigger job D if any of A, B or C have a new successful run since last sensor execution.
Am I understanding correctly ? If that's the case, is there any other way to implement the rule I described ?

Thanks for your help !

**Reply by gibsondan** [reply]
Your understanding is correct - would you mind creating a new discussion for that question about run status sensors? That will help us ensure the right people see it.

**Comment by gibsondan** [comment]
For creating dependencies between jobs, see: https://github.com/dagster-io/dagster/discussions?discussions_q=is%3Aopen+dependency+between+jobs

You can use run status sensors (https://docs.dagster.io/concepts/partitions-schedules-sensors/sensors#run-status-sensors) to kick off one job when another job finishes, or asset sensors (https://docs.dagster.io/concepts/partitions-schedules-sensors/asset-sensors#asset-sensors) to trigger a job or an asset materialization (in a separate run) when assets are materialized.
